File: Python/2674.py
def ehPrimo(n):
    if n <= 3:
        return n > 1
    if not n % 2 or not n % 3:
        return False
    
    i = 5
    
    stop = int(n**0.5)
    
    while i <= stop:
        if not n%i or not n%(i+2):
            return False
        i += 6
    
    return True

while True:
    try:
        
        n = input()      
        primo = ehPrimo(int(n))
        super = False
        
        if primo:
            for i in range(len(n)):
                super = ehPrimo(int(n[i]))
                if not super:
                    break
            if super:
                print("Super")
            else:
                print("Primo")
        else:
            print("Nada")
        
    except EOFError:
        break

File: C/2674.c
#include <stdio.h>
#include <math.h>

int ehPrimo(int n);

int main(){

    int n;

    while(scanf("%i", &n) != EOF){
        int primo = ehPrimo(n), super = 0;

        if(primo == 1){
            do {
                int digito = n % 10;
                super = ehPrimo(digito);
                if(super == 0){
                    break;
                }
            } while (n/=10);

            if(super == 1){
                printf("Super\n");
            }else{
                printf("Primo\n");
            }
        }else{
            printf("Nada\n");
        }
    }

    return 0;
}

int ehPrimo(int n){
    if (n <= 3){
        return n > 1;
    }

    if(n%2 == 0 || n%3 == 0){
        return 0;
    }

    int i = 5, stop = sqrt(n);

    while(i <= stop){
        if(n%i == 0 || n%(i+2) == 0){
            return 0;
        }
        i += 6;
    }

    return 1;

}